resolwe/storage/manager.py

```diff
@@ -46,23 +55,37 @@ def process_delete(self):
                     file_storage = self._lock_file_storage(file_storage.id)
                     if file_storage is None:
                         continue
-                    logger.debug(
-                        "Deleting data from location {}".format(connector_name)
-                    )
-                    try:
-                        self.delete_single_location(file_storage, connector_name)
-                    except Exception:
-                        logger.exception(
-                            "Error deleting data from StorageLocation instance",
-                        )
+                    # Some other storage manager might have already processed this storage
+                    # location before we did.
+                    if not file_storage.storage_locations.filter(
+                        connector_name=connector_name,
+                        url=file_storage.subpath,
+                        status=StorageLocation.STATUS_DONE,
+                    ).exists():
+                        try:
+                            self.delete_single_location(file_storage, connector_name)
+                        except:
+                            logger.exception(
+                                "Error deleting data from StorageLocation instance",
+                            )
 
     def copy_single_location(self, file_storage: FileStorage, connector_name: str):
         """Copy given location to a given connector."""
+
+        # Some other storage manager might have already processed this storage
+        # location before we did.
+        if file_storage.storage_locations.filter(
+            connector_name=connector_name,
+            url=file_storage.subpath,
+            status=StorageLocation.STATUS_DONE,
+        ).exists():
```

Comment:
Same as above. You should create the object instead of just checking that it does not exist.